user-auth/enclave/src/lib.rs

```diff
@@ -0,0 +1,161 @@
+#![no_std]
+#![deny(unsafe_op_in_unsafe_fn)]
+
+use std::slice;
+
+use argon2::{Algorithm, Argon2, Params, PasswordHasher, PasswordVerifier, Version};
+use once_cell::unsync::OnceCell;
+use password_hash::PasswordHashString;
+use sgx_tkeys::{SgxKeyPolicy, SgxSecret, SgxSecretBuilder};
+use sgx_tstd as std;
+use sgx_types::*;
+
+/// Length of a PHC-encoding of a 16 bit salt
+pub const SALT_LENGTH: usize = 22;
+
+/// It is assumed that the hash parameters and other configuration options (e.g.
+/// output length) are all set to their default values.  If any of these are
+/// changed in the future then this value will either need to be redetermined or
+/// dynammically calculated!
+// XXX: Certain assumptions are made here that relate to the hash context. See
+// the doc comment above.
+pub const HASH_STRING_LENGTH: usize = 97;
+
+/// Argon2 memory cost in MiB.
+pub const M_COST_MIB: u32 = 19;
+
+// Convert the Argon2 memory cost from MiB to KiB
+const fn m_cost_calc(mebibytes: u32) -> {
+    mebibytes * 1024
+}
```

Comment:
Just do this in a constant called `M_COST_KIB` and remove `M_COST_MIB`.  Much simpler.